doc/source/overview/high_level.md

```diff
@@ -0,0 +1,508 @@
+\right]dz_1 - c_1
+$$
+
+The term within the integral computes the expectation of the model derivative in $x_1$ over the random variable $X_2$ conditional on $X_1=z_1$. By taking the expectation for $X_2$, we factor out its dependency. So now we know how the prediction $f$ changes local to a point $X_1=z_1$ independent of $X_2$. Integrating these changes over $x_1$ from a minimum value to the value of interest, we obtain the global plot of how the model depends on $x_1$. ALE-plots get their names as they accumulate (integrate) the local effects (the expected partial derivatives). Note that here we have assumed $f$ is differentiable. In practice, however, we compute the various quantities above numerically, so this isn't a requirement.
+
+__TODO__:
+- Add picture explaining the above idea.
+
+For more details on accumulated local effects including a discussion on PDP-plots and M-plots see
+[Motivation-and-definition for ALE](../methods/ALE.ipynb#Motivation-and-definition)
+
+:::{admonition} **Note 4: Categorical Variables and ALE**
+Note that because ALE plots require computing differences between variables, they don't naturally extend to categorical data unless there is a sensible ordering on the data. As an example, consider the months of the year. To be clear, this is only an issue if the variable you are taking the ALE for is categorical.
+:::
+
+**Pros**:
+- ALE-plots are easy to visualize and understand intuitively
+- Very general as it is a black box algorithm
+- Doesn't struggle with dependencies in the underlying features, unlike PDP plots
+- ALE plots are fast
+
+**Cons**:
+- Harder to explain the underlying motivation behind the method than PDP plots or M plots.
+- Requires access to the training dataset.
+- Unlike PDP plots, ALE plots do not work with Categorical data
+
+### Counter Factuals:
+
+Given an instance of the dataset and a prediction given by a model, a question naturally arises how would the instance minimally have to change for a different prediction to be provided. Counterfactuals are local explanations as they relate to a single instance and model prediction.
+
+Given a classification model trained on the MNIST dataset and a sample from the dataset with a given prediction, a counterfactual would be a generated image that closely resembles the original but is changed enough that the model correctly classifies it as a different number.
+
+__TODO__:
+- Give example image to illustrate
+
+Similarly, given tabular data that a model uses to make financial decisions about a customer, a counterfactual would explain how to change their behavior to obtain a different conclusion. Alternatively, it may tell the Machine Learning Engineer that the model is drawing incorrect assumptions if the recommended changes involve features that are irrelevant to the given decision.
+
+A counterfactual, $x_{cf}$, needs to satisfy
+
+- The model prediction on $x_{cf}$ needs to be close to the predefined output.
```

Comment:
predefined --> pre-defined

Reply:
I would also be more explicit, e.g. "pre-defined output (e.g. desired class label)."